src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/ReduceOutputDocumentActions.cs

```diff
@@ -51,6 +65,34 @@ public void Initialize(RavenTransaction tx)
             }
         }
 
+        public void AddPatternGeneratedIdForReduceOutput(Transaction tx, string reduceResultId, string referenceDocumentId)
+        {
+            var tree = tx.CreateTree(ReduceOutputsIdsToPatternReferenceIdsTreeName);
```

Comment:
shouldn't we cache that?

Reply:
`Transaction` has it cached already. I just changed `string ReduceOutputsIdsToPatternReferenceIdsTreeName` to `Slice` so we won't cause additional allocations